definitions/features/sync_plans.rb

```diff
@@ -3,33 +3,49 @@ class Features::SyncPlans < ForemanMaintain::Feature
     label :sync_plans
   end
 
-  def active_sync_plans_count
-    feature(:foreman_database).query(
-      <<-SQL
-        SELECT count(*) AS count FROM katello_sync_plans WHERE enabled ='t'
-      SQL
-    ).first['count'].to_i
+  def required_new_implementation
+    @required_new_implementation ||=
+      feature(:foreman_database).query(
+        <<-SQL
+          SELECT COUNT(1) FROM information_schema.table_constraints
+          WHERE constraint_name='katello_sync_plan_foreman_tasks_recurring_logic_fk' AND table_name='katello_sync_plans'
+        SQL
+      ).first['count'].to_i > 0
   end
 
   def ids_by_status(enabled = true)
-    enabled = enabled ? 't' : 'f'
-    feature(:foreman_database).query(
-      <<-SQL
-        SELECT id FROM katello_sync_plans WHERE enabled ='#{enabled}'
+    if required_new_implementation
```

Comment:
@kgaikwad `ids_by_status` and `verify_existing_ids_by_status` are almost identical inside, could you please refator?